django/db/backends/postgresql/schema.py

```diff
@@ -94,3 +94,43 @@ def _alter_column_type_sql(self, table, old_field, new_field, new_type):
             return super(DatabaseSchemaEditor, self)._alter_column_type_sql(
                 table, old_field, new_field, new_type
             )
+
+    def _alter_field(self, model, old_field, new_field, old_type, new_type,
+                     old_db_params, new_db_params, strict=False):
+        super(DatabaseSchemaEditor, self)._alter_field(
+            model, old_field, new_field, old_type, new_type, old_db_params,
+            new_db_params, strict
+        )
+        # Added an index?
+        if ((not old_field.db_index and new_field.db_index) or (not old_field.unique and new_field.unique)):
+            db_type = new_field.db_type(connection=self.connection)
+            if db_type is not None and (new_field.db_index or new_field.unique):
```

Comment:
Let's factor the shared logic from `_model_indexes_sql()` so we don't repeat it all.